File: 2161-partition-array-according-to-given-pivot/2161-partition-array-according-to-given-pivot.cpp
class Solution {
public:
    vector<int> pivotArray(vector<int>& nums, int pivot) {
        
        vector<int> small;
        vector<int> big;
        vector<int> eql;
        
        for (auto a : nums) {
            if (pivot > a) {
                small.push_back(a);
            } else if (pivot < a) {
                big.push_back(a);
            } else {
                eql.push_back(a);
            }
        }
        vector<int> ans;
        for (auto a : small)  ans.push_back(a);
        for (auto a : eql)  ans.push_back(a);
        for (auto a : big)  ans.push_back(a);
        
        
        return ans;
    }
};

File: 2161-partition-array-according-to-given-pivot/2161-partition-array-according-to-given-pivot.java
class Solution {
    public int[] pivotArray(int[] nums, int pivot) {
        
        
        ArrayList<Integer> less= new ArrayList<>();
         ArrayList<Integer> equal= new ArrayList<>();
         ArrayList<Integer> great= new ArrayList<>();
       ArrayList<Integer> ans= new ArrayList<>();
        
        for(int i=0;i<nums.length;i++){
            
            if(nums[i]==pivot){
                equal.add(nums[i]);
            }
            
            else if(nums[i]<pivot){
                less.add(nums[i]);
            }
            else 
            {
                great.add(nums[i]);
            }
            
        }
        
      for(Integer e : less){
          ans.add(e);
      }
        
         for(Integer e : equal){
          ans.add(e);
      }
        
         for(Integer e : great){
          ans.add(e);
      }
        
        int[] res = new int[nums.length];
        
        for(int i=0;i<nums.length;i++){
            res[i]=ans.get(i);
        }
        
        return res;
       
        
    }
}